code/game/objects/items/devices/oxycandle.dm

```diff
@@ -0,0 +1,72 @@
+/obj/item/device/oxycandle
+	name = "Oxygen Candle"
+	desc = "Oxygen candles are flare-like steel tubes of sodium chlorate that produces oxygen through chemical process."
+	icon = 'icons/obj/candle.dmi'
+	icon_state = "candle1"
+	w_class = ITEMSIZE_SMALL // Should fit into internal's box or maybe pocket
+	var/target_pressure = ONE_ATMOSPHERE
+	var/datum/gas_mixture/air_contents = null
+	var/volume = 5500 //One tile has 2500 volume of air, so two tiles plus a bit extra
+	var/on = 0
+	var/activation_sound = 'sound/items/flare.ogg'
+	light_color = LIGHT_COLOR_FLARE
+	light_wedge = LIGHT_OMNI
+	uv_intensity = 50
+	var/brightness_on = 2 // Moderate bright.
+	light_power = 2
+	action_button_name = null
+
+/obj/item/device/oxycandle/attack_self(mob/user)
+	if(!on)
+		light_range = brightness_on
+		on = 1
+		update_icon()
+		playsound(src.loc, activation_sound, 75, 1)
+		air_contents = new /datum/gas_mixture()
+		air_contents.volume = 200 //liters
+		air_contents.temperature = T20C
+		var/list/air_mix = StandardAirMix()
+		air_contents.adjust_multi("oxygen", air_mix["oxygen"], "nitrogen", air_mix["nitrogen"])
+		START_PROCESSING(SSprocessing, src)
+	else
+		var/turf/pos = get_turf(src)
+		if(pos)
+			pos.hotspot_expose(1500, 5)
```

Comment:
If it's on, you can keep using it to `hotspot_expose`?

Reply:
Since it is chemical reactor, and it is burning we make it light up phoron and fuel on tiles!